File: frontend/src/types/payment.ts
export interface Payment {
  id: number;
  bookingId: number;
  amount: number;
  paymentType: 'DEPOSIT' | 'REMAINING' | 'FULL';
  paymentMethod: 'CARD' | 'CASH';
  status: 'PENDING' | 'COMPLETED' | 'FAILED' | 'REFUNDED';
  createdAt: string;
  completedAt?: string;
}

export interface PaymentIntentRequest {
  bookingId: number;
  paymentType: 'DEPOSIT' | 'REMAINING' | 'FULL';
  amount: number;
}

export interface PaymentIntentResponse {
  clientSecret: string;
  paymentId: number;
  paymentIntentId: string;
}

export interface StripeConfig {
  publishableKey: string;
}

File: backend/src/main/java/ro/atm/backend/domain/payment/entity/Payment.java
package ro.atm.backend.domain.payment.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import ro.atm.backend.domain.booking.entity.Booking;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    public enum PaymentStatus {
        PENDING,
        COMPLETED,
        FAILED,
        REFUNDED
    }

    public enum PaymentType {
        DEPOSIT,
        REMAINING,
        FULL
    }

    public enum PaymentMethod {
        CARD,
        CASH
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "booking_id", nullable = false)
    private Booking booking;

    @Column(nullable = false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentType paymentType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentMethod paymentMethod;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus status = PaymentStatus.PENDING;

    @Column(unique = true)
    private String stripePaymentIntentId;

    @Column(unique = true)
    private String stripeClientSecret;

    @CreationTimestamp
    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime completedAt;

    @Column(length = 1000)
    private String failureReason;
}
